concent_api/core/tasks.py

```diff
@@ -2,10 +2,83 @@
 
 from celery import shared_task
 
+from conductor.tasks import upload_acknowledged
+from core.models import PendingResponse
+from core.models import Subtask
+from core.payments import base
+from core.subtask_helpers import update_subtask_state
+from core.transfer_operations import store_pending_message
+from utils.helpers import deserialize_message
+from utils.helpers import get_current_utc_timestamp
+
 
 logger = logging.getLogger(__name__)
 
 
 @shared_task
-def upload_finished(_subtask_id: str):
-    pass
+def upload_finished(subtask_id: str):
```

Comment:
too many reruns in this function... it makes it a bit hard to read. Please refactor

Reply:
done

Reply:
done

Reply:
Actually it was done on purpose for two reasons: 
1. To make the task easier to read. It was showing clearly where and when the task should end. 
2. To avoid future errors. Now, if somebody put some code at the end of the task and forget to add returns everywhere where they were before, the task will not work correctly.

Please add them back.
I even think we should add this as a good practice for celery tasks.